index.bs

```diff
@@ -2443,74 +2325,116 @@ partial interface MLGraphBuilder {
   </div>
 </details>
 
-<details open>
+#### add #### {#api-mlgraphbuilder-binary-add}
+Add the values of the two input tensors, element-wise. See [[#api-mlgraphbuilder-binary]] for more detail.
+
+#### div #### {#api-mlgraphbuilder-binary-div}
+Divide the values of the first input tensor with the values of the second tensor, element-wise. See [[#api-mlgraphbuilder-binary]] for more detail.
+
+#### max #### {#api-mlgraphbuilder-binary-max}
+Select the greater values of the two input tensors, element-wise. See [[#api-mlgraphbuilder-binary]] for more detail.
+
+#### min #### {#api-mlgraphbuilder-binary-min}
+Select the lesser values of the two input tensors, element-wise. See [[#api-mlgraphbuilder-binary]] for more detail.
+
+#### mul #### {#api-mlgraphbuilder-binary-mul}
+Multiply the values of the two input tensors, element-wise. See [[#api-mlgraphbuilder-binary]] for more detail.
+
+#### pow #### {#api-mlgraphbuilder-binary-pow}
+Compute the values of the values of the first input tensor to the power of the values of the second input tensor, element-wise. See [[#api-mlgraphbuilder-binary]] for more detail.
+
+#### sub #### {#api-mlgraphbuilder-binary-sub}
+Subtract the values of the second input tensor from the values of the first input tensor, element-wise. See [[#api-mlgraphbuilder-binary]] for more detail.
+
+### Element-wise logical operations ### {#api-mlgraphbuilder-logical}
+Compare two input tensors element-wise, if the operation requires two inputs, and return a tensor of Boolean values 0 or 1 for the specified input values.
+
+The two input tensors will be broadcasted according to [[!numpy-broadcasting-rule]]. The [=rank=] of the output tensor is the maximum
+[=rank=] of the input tensors.
+
+<script type=idl>
+partial interface MLGraphBuilder {
+  MLOperand equal(MLOperand a, MLOperand b);
+  MLOperand greater(MLOperand a, MLOperand b);
+  MLOperand greaterOrEqual(MLOperand a, MLOperand b);
+  MLOperand lesser(MLOperand a, MLOperand b);
+  MLOperand lesserOrEqual(MLOperand a, MLOperand b);
+  MLOperand not(MLOperand a);
```

Comment:
(oops, I wrote my comment in the wrong place earlier - I'll shorten it here and split my thoughts so the main points are hopefully clearer)

There are "bitwise" operators and "logical" operators, and I only introduced the minimum for the pertinent models rather than the full picture, meaning we haven't introduced the bitwise ops yet, but *eventually* we should, as every mature ML library has them:

| Logical ops | Bitwise ops |
|--------------|-------------|
| torch.logical_not | torch.bitwise_not |
| torch.logical_and | torch.bitwise_and |
| torch.logical_or | torch.bitwise_or |
| torch.logical_xor | torch.bitwise_xor |
| - | torch.bitwise_left_shift |
| - | torch.bitwise_right_shift |
| tf.math.logical_and | tf.bitwise.bitwise_and |
| tf.math.logical_or | tf.bitwise.bitwise_or |
| tf.math.logical_xor | tf.bitwise.bitwise_xor |
| - | tf.bitwise.left_shift |
| - | tf.bitwise.right_shift |
| numpy.logical_not | numpy.invert |
| numpy.logical_and | numpy.bitwise_and |
| numpy.logical_or | numpy.bitwise_or |
| numpy.logical_xor | numpy.bitwise_xor |
| - | numpy.left_shift |
| - | numpy.right_shift |
| [tosa.LOGICAL_NOT](https://www.mlplatform.org/tosa/tosa_spec.html#_logical_not) | [tosa.BITWISE_NOT](https://www.mlplatform.org/tosa/tosa_spec.html#_bitwise_not) |
| tosa.LOGICAL_AND | tosa.BITWISE_AND |
| tosa.LOGICAL_OR | tosa.BITWISE_OR |
| tosa.LOGICAL_AND | tosa.BITWISE_AND |
| tosa.LOGICAL_XOR | tosa.BITWISE_XOR |
| - | tosa.LOGICAL_LEFT_SHIFT |
| - | tosa.LOGICAL_RIGHT_SHIFT |
| DML_ELEMENT_WISE_LOGICAL_NOT | DML_ELEMENT_WISE_BIT_NOT |
| DML_ELEMENT_WISE_LOGICAL_AND | DML_ELEMENT_WISE_BIT_AND |
| DML_ELEMENT_WISE_LOGICAL_OR | DML_ELEMENT_WISE_BIT_OR |
| DML_ELEMENT_WISE_LOGICAL_XOR | DML_ELEMENT_WISE_BIT_XOR |
| - | DML_ELEMENT_WISE_BIT_SHIFT_LEFT |
| - | DML_ELEMENT_WISE_BIT_SHIFT_RIGHT |

Because most of these operators (except the shifts) have both "logical" and "bitwise" counterparts, shortening names (e.g. "bitwiseAnd" or "logicalAnd" to just "and") leads to ambiguity of *which* of the two "and"'s is meant. That's probably why most libraries spell them out to disambiguate. Yes, ONNX chose the shorter form `And` to mean a `LogicalAnd`, but it just as easily could have blessed `BitwiseAnd` instead.

Really I introduced `not` because the `notEqual` operator (as found in ([PyTorch not_equal](https://pytorch.org/docs/stable/generated/torch.not_equal.html), [TensorFlow not_equal](https://www.tensorflow.org/api_docs/python/tf/math/not_equal), [TOSA compare NE](https://github.com/openxla/stablehlo/blob/main/docs/spec.md#compare)) is missing from ONNX, which composes `notEqual` from `equal`+`not`. If we had a true `notEqual`, then having logical not wouldn't even matter, as WebNN EP could map equal+not -> notEqual. What do you think?

Reply:
@fdwr please feel free to open a new issue as is deemed appropriate.

Resolving this conversation.